File: merg_sort/mergSort.go
package main

import "strings"

func main() {
	f := "arshia"
	r := merg_sort(f)
	println(r)

}

func merg_sort(arr string) string {
	if len(arr) < 2 {
		return arr
	}
	mid := len(arr) / 2
	left := merg_sort(arr[:mid])
	right := merg_sort(arr[mid:])
	return merg(left, right)
}

func merg(left, right string) string {
	i := 0
	j := 0
	var result []string
	for len(left) > i && len(right) > j {
		if left[i] < right[j] {
			result = append(result, string(left[i]))
			i++
		} else {
			result = append(result, string(right[j]))
			j++
		}
	}
	result = append(result, string(left[i:]))
	result = append(result, string(right[j:]))
	return strings.Join(result,"")

}

File: merg_sort/mergSort.py
def merg_sort(s):
    if len(s) < 2:
        return s
    mid = len(s) // 2
    left = merg_sort(s[:mid])
    right = merg_sort(s[mid:])
    return merg(left, right)


def merg(left, right):
    result = []
    i = 0
    j = 0
    print(left)
    while len(left) > i and len(right) > j:
        if left[i] < right[j]:
            result.append(left[i])
            i += 1
        else:
            result.append(right[j])
            j += 1
    result += left[i:]
    result += right[j:]
    return "".join(result)

print(merg_sort("arshia"))
    